dash/development/_r_components_generation.py

```diff
@@ -529,38 +550,16 @@ def generate_rpkg(
     # generate the internal (not exported to the user) functions which
     # supply the JavaScript dependencies to the dash package.
     # this avoids having to generate an RData file from within Python.
-    write_js_metadata(pkg_data=pkg_data, project_shortname=project_shortname)
+    write_js_metadata(pkg_data, project_shortname, has_wildcards)
 
-    with open("NAMESPACE", "w") as f:
+    with open("NAMESPACE", "w+") as f:
```

Comment:
🐮  minor it's personal preference, but I would avoid `f` with numbers, maybe for example fp_ns, fp_desr